### PR TITLE
[Static web assets] Additional unit tests and fixes

### PR DESCRIPTION
**Pull request by javiercn**
* Adds unit tests for most of the MSBuild tasks.
* Validate that all asset paths are unique when we build the manifest.
* Fixes a couple of small bugs when we grouped assets by Identity instead of `target path`.
* Fixes the way we merge properties for projects to account for `UndefineProperties`

### CODE REVIEW

**Comment by javiercn** [comment]
Fixes https://github.com/dotnet/sdk/issues/19704

File: src/Tests/Microsoft.NET.Sdk.Razor.Tests/AspNetSdkBaselineTest.cs [hunk 1]
```diff
@@ -23,7 +23,7 @@ public class AspNetSdkBaselineTest : AspNetSdkTest
 #if GENERATE_SWA_BASELINES
         public static bool GenerateBaselines = true;
 #else
-        public static bool GenerateBaselines = false;
+        public static bool GenerateBaselines = true;
```
Comment (pranavkm) [inline src/Tests/Microsoft.NET.Sdk.Razor.Tests/AspNetSdkBaselineTest.cs:26]:
```suggestion
        public static bool GenerateBaselines = false;
```

We should add a test for this that says that `Assert.False(GenerateBaselines);`

File: src/Tests/Microsoft.NET.Sdk.Razor.Tests/StaticWebAssets/ReadStaticWebAssetsManifestFileTest.cs [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+            task.ReferencedProjectsConfiguration.Should().BeNull();
+        }
+
+        public void Dispose()
+        {
+            try
+            {
+                if (TempFilePath != null && File.Exists(TempFilePath))
+                {
+                    File.Delete(TempFilePath);
```
Comment (pranavkm) [inline src/Tests/Microsoft.NET.Sdk.Razor.Tests/StaticWebAssets/ReadStaticWebAssetsManifestFileTest.cs:338]: The SDK tests leave these files around so you can see it. They write to the Intermediate dir, and use the same file name for each run, so the number of files doesn't grow. Perhaps we could consider doing the same thing here.

**Review by pranavkm — APPROVED** [review]

File: src/Tests/Microsoft.NET.Sdk.Razor.Tests/AspNetSdkBaselineTest.cs (hunk 1 above)
Comment (javiercn) [reply src/Tests/Microsoft.NET.Sdk.Razor.Tests/AspNetSdkBaselineTest.cs]: Good catch, I just did.

File: src/Tests/Microsoft.NET.Sdk.Razor.Tests/StaticWebAssets/ReadStaticWebAssetsManifestFileTest.cs (hunk 2 above)
Comment (javiercn) [reply src/Tests/Microsoft.NET.Sdk.Razor.Tests/StaticWebAssets/ReadStaticWebAssetsManifestFileTest.cs]: I changed to use the TestContext and put the temp files in the temp path.

**Pull request merged**